vendor/ddcloud/resource_additional_nics_test.go

```diff
@@ -0,0 +1,294 @@
+    }
+	`, name, description, primaryIPv4Address)
+}
+
+// add a nic to the server with ipv4address as input and verify that it gets created with the correct configuration.
+func TestAccServerAdditionalNicCreateWithIPV4Address(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testCheckDDCloudAdditionalNicDestroy,
+			testCheckDDCloudServerDestroy,
+			testCheckDDCloudVLANDestroy,
+			testCheckDDCloudNetworkDomainDestroy,
+		),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDDCloudAdditionalNicToServerUsingIPV4Address(
+					"acc-test-server",
+					"Server for Terraform acceptance test.",
+					"192.168.17.11",
+					"192.168.18.100",
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckDDCloudServerExists("ddcloud_server.acc_test_server", true),
+					testCheckDDCloudAdditionalNicMatchesIPV4("ddcloud_server.acc_test_server",
+						"192.168.18.100",
+					),
+				),
+			},
+		},
+	})
+}
+
+// add a nic to the server with ipv4address as input and verify that it gets created with the correct configuration.
+func TestAccServerAdditionalNicCreateWithVLANID(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testCheckDDCloudAdditionalNicDestroy,
+			testCheckDDCloudServerDestroy,
+			testCheckDDCloudVLANDestroy,
+			testCheckDDCloudNetworkDomainDestroy,
+		),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDDCloudAdditionalNicToServerUsingVLANID(
+					"acc-test-server",
+					"Server for Terraform acceptance test.",
+					"192.168.17.11",
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckDDCloudServerExists("ddcloud_server.acc_test_server", true),
+					testCheckDDCloudAdditionalNicMatchesVLANID("ddcloud_server.acc_test_server",
+						"ddcloud_vlan.acc_test_vlan1",
+					),
+				),
+			},
+		},
+	})
+}
+
+// Check if the additional nic configuration matches the expected configuration.
+func testCheckDDCloudAdditionalNicMatchesIPV4(serverResourceName string, expected string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		serverResource, ok := state.RootModule().Resources[serverResourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", serverResourceName)
+		}
+
+		serverID := serverResource.Primary.ID
+
+		client := testAccProvider.Meta().(*providerState).Client()
+		server, err := client.GetServer(serverID)
+		if err != nil {
+			return fmt.Errorf("Bad: Get server: %s", err)
+		}
+		if server == nil {
+			return fmt.Errorf("Bad: Server not found with Id '%s'", serverID)
+		}
+
+		if len(server.Network.AdditionalNetworkAdapters) == 0 {
+			return fmt.Errorf("Bad: Server '%s' has no additional nics", serverID)
+		}
+
+		if *server.Network.AdditionalNetworkAdapters[0].PrivateIPv4Address != expected {
```

Comment:
Can I suggest maybe making the index a parameter of the function, or making the function check **all** NICs for a match?